### DISCUSSION TITLE
Uncaught ReferenceError: calcularChinchilas is not defined     at HTMLButtonElement.onclick (ex5_9.html:20:44)<title>

### DISCUSSION THREAD

**Discussion by lnr2289** [community/community#72071] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Olá a todos, criei um código em html e javascript, alguém pode me ajudar a encontrar e corrigir o erro acima?

<!DOCTYPE html>
<html lang="pt-br">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Exemplo 5.9</title>
</head>

<body>
    <h1>Programa Criação de Chinchilas</h1>
    <form></form>
    <p>Informe o número inicial de Chinchilas e o número de anos:</p>
    <label for="numeroInicial">Número Inicial de Chinchilas:</label>
    <input type="number" id="numeroInicial" min="2" step="1" required>
    <p>
        <label for="anos">Número de Anos:</label>
        <input type="number" id="anos" min="1" step="1" required>
    </p>
    <button onclick="calcularChinchilas()">Calcular</button>
    </form>
    <h3></h3>
    <script src="ex5_9.js"></script>
</body>
</html>

//Código JS
const frm = document.getElementById("form")
const resp = document.getElementById("h3")

frm.addEventListener("submit", (e) => {
    e.preventDefault()
        < script >
        function calcularCrescChinchilas() {
            const numeroInicial = parseInt(document.getElementById("numeroInicial").value)
            const anos = parseInt(document.getElementById("anos").value)
            const resultado []
            if (numeroInicial < 2) {
                alert("O número inicial deve ser maior ou igual a 2 (um casal).")
                return;
            }
            if (anos <= 0) {
                document.getElementById("resultado").innerHTML = "Insira um número de anos válido.";
                return;
            }
            for (var i = 1; i <= anos i++) {
                if (i === 1) {
                    resultado.push("Ano 1: " + numeroInicial + " chinchilas");
                } else {
                    numeroInicial *= 3;
                    resultado.push("Ano " + i + ": " + numeroInicial + " chinchilas");
                }
            }
        }
    document.getElementById("resultado").innerHTML = resultado.join("<br>");
}
    <script />

**Comment by ironest** [comment]
Hello!
The error message complains about not finding `calcularChinchilas`.
In the code you provided, there's a function named `calcularCrescChinchilas`.
Perhaps, that's the problem? Just a simple typo?

¡Hola!
El mensaje de error se queja de no encontrar `calcularChinchilas`.
En el código que proporcionaste, hay una función llamada `calcularCrescChinchilas`.
¿Quizás ese sea el problema? ¿Solo un simple error tipográfico?
